src/workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,134 @@
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# p5 Visualizer Template
+
+[image should go here]
+
+This template can be used as a starting point for designing a p5 visualizer.
+
+## Parameters
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    name = 'p5 Visualizer Template' // the name shown on the visualizer menu
+
+    // external parameters, which the user can choose while creating the
+    // visualizer bundle
+    baseIndex = 0 // the default value of this parameter is zero
+    params = {
+        /** md
+- baseIndex: Where in the sequence to start browsing.
+         **/
+        baseIndex: {
+            value: this.baseIndex,
+            displayName: 'Base index',
+            required: true,
+        },
+    }
+
+    // internal parameters. the value of `index` will always be overwritten in
+    // `setup`, but TypeScript can't infer that, so we have to give `index` an
+    // initial value
+    index = 0
+    flash = 0
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the base index doesn't precede the first index
+        if (this.params.baseIndex.value < this.seq.first) {
```

Comment:
This unfortunately doesn't work the way you want it to. (A) the sequence may not be set at the time that parameters are being manipulated by the user, and (B) the ultimate design is that we want to allow the sequence to change on the fly. So at the moment, any checks against the sequence need to come in inhabit() or later, and setup() is probably better (because that's even just a bit later).

If you want to give an example of a validation that could appropriately go in here, I suppose you could artificially limit the start index to 9999 (since almost all sequences in the OEIS only have 10000 terms). Can't really think of any other validation that would make sense. Even that's a little misleading, because there is no real reason to limit the startIndex. I think it's pretty tough to think of a reasonable validation on just a single sequence index. If you added an upper limit on browsing, you could check if they were in the correct order.  Anyhow, the example can't really work as is with the current design, so it needs to change.

Reply:
>  (A) the sequence may not be set at the time that parameters are being manipulated by the user,

Good point—I've emphasized this in the visualizer-building guide. In the example, I've replaced the base index parameter with a step size parameter, which has to be a positive integer.

Reply:
OK, access of sequence in setup() looks good now.